<!DOCTYPE html>


<html theme="dark" showBanner="true" hasBanner="true" > 
<link href="https://cdn.staticfile.org/font-awesome/6.4.2/css/fontawesome.min.css" rel="stylesheet">
<link href="https://cdn.staticfile.org/font-awesome/6.4.2/css/brands.min.css" rel="stylesheet">
<link href="https://cdn.staticfile.org/font-awesome/6.4.2/css/solid.min.css" rel="stylesheet">
<link href="https://lf9-cdn-tos.bytecdntp.com/cdn/expire-1-M/KaTeX/0.10.2/katex.min.css" rel="stylesheet">
<script src="/js/color.global.min.js" ></script>
<script src="/js/load-settings.js" ></script>
<head>
  <meta charset="utf-8">
  
  
  

  
  <title>Seqkit Usage | Jichen&#39;s Blog</title>
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="preload" href="/css/fonts/Roboto-Regular.ttf" as="font" type="font/ttf" crossorigin="anonymous">
  <link rel="preload" href="/css/fonts/Roboto-Bold.ttf" as="font" type="font/ttf" crossorigin="anonymous">

  <meta name="description" content="seqkit usage  Repo直达 https:&#x2F;&#x2F;github.com&#x2F;shenwei356&#x2F;seqkit   Introduction seqkit是重庆医科大学病毒性肝炎研究所副研究员沈伟老师基于Go语言开发的跨平台的、快速的、全面的fasta&#x2F;q处理工具  Installation 1234# Generalconda install -c biconda seqkit# Alte">
<meta property="og:type" content="article">
<meta property="og:title" content="Seqkit Usage">
<meta property="og:url" content="http://example.com/2023/10/16/seqkit_usage/index.html">
<meta property="og:site_name" content="Jichen&#39;s Blog">
<meta property="og:description" content="seqkit usage  Repo直达 https:&#x2F;&#x2F;github.com&#x2F;shenwei356&#x2F;seqkit   Introduction seqkit是重庆医科大学病毒性肝炎研究所副研究员沈伟老师基于Go语言开发的跨平台的、快速的、全面的fasta&#x2F;q处理工具  Installation 1234# Generalconda install -c biconda seqkit# Alte">
<meta property="og:locale" content="zh_CN">
<meta property="og:image" content="http://example.com/2023/10/16/seqkit_usage/image-1.png">
<meta property="og:image" content="http://example.com/2023/10/16/seqkit_usage/image-2.png">
<meta property="og:image" content="http://example.com/2023/10/16/seqkit_usage/image-3.png">
<meta property="og:image" content="http://example.com/2023/10/16/seqkit_usage/image-4.png">
<meta property="og:image" content="http://example.com/2023/10/16/seqkit_usage/image-5.png">
<meta property="og:image" content="http://example.com/2023/10/16/seqkit_usage/image-6.png">
<meta property="og:image" content="http://example.com/2023/10/16/seqkit_usage/image-7.png">
<meta property="og:image" content="http://example.com/2023/10/16/seqkit_usage/image-8.png">
<meta property="og:image" content="http://example.com/2023/10/16/seqkit_usage/image-9.png">
<meta property="og:image" content="http://example.com/2023/10/16/seqkit_usage/image-10.png">
<meta property="og:image" content="http://example.com/2023/10/16/seqkit_usage/image-11.png">
<meta property="article:published_time" content="2023-10-15T19:07:08.000Z">
<meta property="article:modified_time" content="2023-11-04T12:51:21.285Z">
<meta property="article:author" content="Jichen Wen">
<meta property="article:tag" content="软件使用">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="http://example.com/2023/10/16/seqkit_usage/image-1.png">
  
    <link rel="alternate" href="/atom.xml" title="Jichen's Blog" type="application/atom+xml">
  
  
    <link rel="icon" media="(prefers-color-scheme: light)" href="/images/favicon-light-32.png" sizes="32x32">
    <link rel="icon" media="(prefers-color-scheme: light)" href="/images/favicon-light-128.png" sizes="128x128">
    <link rel="icon" media="(prefers-color-scheme: light)" href="/images/favicon-light-180.png" sizes="180x180">
    <link rel="icon" media="(prefers-color-scheme: light)" href="/images/favicon-light-192.png" sizes="192x192">
    <link rel="icon" media="(prefers-color-scheme: dark)" href="/images/favicon-dark-32.png" sizes="32x32">
    <link rel="icon" media="(prefers-color-scheme: dark)" href="/images/favicon-dark-128.png" sizes="128x128">
    <link rel="icon" media="(prefers-color-scheme: dark)" href="/images/favicon-dark-180.png" sizes="180x180">
    <link rel="icon" media="(prefers-color-scheme: dark)" href="/images/favicon-dark-192.png" sizes="192x192">
  
  
<link rel="stylesheet" href="/css/style.css">

<meta name="generator" content="Hexo 6.3.0"></head>

<body>
  
  
    
<div id="banner" class="">
  <img src="/banner.jpg" itemprop="image">
  <div id="banner-dim"></div>
</div>
 
   
  <div id="main-grid" class="shadow   ">
    <div id="nav" class=""  >
      <navbar id="navbar">
  <nav id="title-nav">
    <a href="/">
      <div id="vivia-logo">
        <div class="dot"></div>
        <div class="dot"></div>
        <div class="dot"></div>
        <div class="dot"></div>
      </div>
      <div>Jichen's Blog </div>
    </a>
  </nav>
  <nav id="main-nav">
    
      <a class="main-nav-link" href="/">Home</a>
    
      <a class="main-nav-link" href="/archives">Archives</a>
    
      <a class="main-nav-link" href="/about">About</a>
    
  </nav>
  <nav id="sub-nav">
    <a id="theme-btn" class="nav-icon">
      <span class="light-mode-icon"><svg xmlns="http://www.w3.org/2000/svg" height="20" viewBox="0 -960 960 960" width="20"><path d="M438.5-829.913v-48q0-17.452 11.963-29.476 11.964-12.024 29.326-12.024 17.363 0 29.537 12.024t12.174 29.476v48q0 17.452-11.963 29.476-11.964 12.024-29.326 12.024-17.363 0-29.537-12.024T438.5-829.913Zm0 747.826v-48q0-17.452 11.963-29.476 11.964-12.024 29.326-12.024 17.363 0 29.537 12.024t12.174 29.476v48q0 17.452-11.963 29.476-11.964 12.024-29.326 12.024-17.363 0-29.537-12.024T438.5-82.087ZM877.913-438.5h-48q-17.452 0-29.476-11.963-12.024-11.964-12.024-29.326 0-17.363 12.024-29.537t29.476-12.174h48q17.452 0 29.476 11.963 12.024 11.964 12.024 29.326 0 17.363-12.024 29.537T877.913-438.5Zm-747.826 0h-48q-17.452 0-29.476-11.963-12.024-11.964-12.024-29.326 0-17.363 12.024-29.537T82.087-521.5h48q17.452 0 29.476 11.963 12.024 11.964 12.024 29.326 0 17.363-12.024 29.537T130.087-438.5Zm660.174-290.87-34.239 32q-12.913 12.674-29.565 12.174-16.653-.5-29.327-13.174-12.674-12.673-12.554-28.826.12-16.152 12.794-28.826l33-35q12.913-12.674 30.454-12.674t30.163 12.847q12.709 12.846 12.328 30.826-.38 17.98-13.054 30.653ZM262.63-203.978l-32 34q-12.913 12.674-30.454 12.674t-30.163-12.847q-12.709-12.846-12.328-30.826.38-17.98 13.054-30.653l33.239-31q12.913-12.674 29.565-12.174 16.653.5 29.327 13.174 12.674 12.673 12.554 28.826-.12 16.152-12.794 28.826Zm466.74 33.239-32-33.239q-12.674-12.913-12.174-29.565.5-16.653 13.174-29.327 12.673-12.674 28.826-13.054 16.152-.38 28.826 12.294l35 33q12.674 12.913 12.674 30.454t-12.847 30.163q-12.846 12.709-30.826 12.328-17.98-.38-30.653-13.054ZM203.978-697.37l-34-33q-12.674-12.913-13.174-29.945-.5-17.033 12.174-29.707t31.326-13.293q18.653-.62 31.326 13.054l32 34.239q11.674 12.913 11.174 29.565-.5 16.653-13.174 29.327-12.673 12.674-28.826 12.554-16.152-.12-28.826-12.794ZM480-240q-100 0-170-70t-70-170q0-100 70-170t170-70q100 0 170 70t70 170q0 100-70 170t-170 70Zm-.247-82q65.703 0 111.475-46.272Q637-414.544 637-480.247t-45.525-111.228Q545.95-637 480.247-637t-111.475 45.525Q323-545.95 323-480.247t45.525 111.975Q414.05-322 479.753-322ZM481-481Z"/></svg></span>
      <span class="dark-mode-icon"><svg xmlns="http://www.w3.org/2000/svg" height="20" viewBox="0 -960 960 960" width="20"><path d="M480.239-116.413q-152.63 0-258.228-105.478Q116.413-327.37 116.413-480q0-130.935 77.739-227.435t206.304-125.043q43.022-9.631 63.87 10.869t3.478 62.805q-8.891 22.043-14.315 44.463-5.424 22.42-5.424 46.689 0 91.694 64.326 155.879 64.325 64.186 156.218 64.186 24.369 0 46.978-4.946 22.609-4.945 44.413-14.076 42.826-17.369 62.967 1.142 20.142 18.511 10.511 61.054Q807.174-280 712.63-198.206q-94.543 81.793-232.391 81.793Zm0-95q79.783 0 143.337-40.217 63.554-40.218 95.793-108.283-15.608 4.044-31.097 5.326-15.49 1.283-31.859.805-123.706-4.066-210.777-90.539-87.071-86.473-91.614-212.092-.24-16.369.923-31.978 1.164-15.609 5.446-30.978-67.826 32.478-108.282 96.152Q211.652-559.543 211.652-480q0 111.929 78.329 190.258 78.329 78.329 190.258 78.329ZM466.13-465.891Z"/></svg></span>
    </a>
    
      <a id="nav-rss-link" class="nav-icon mobile-hide" href="/atom.xml" title="RSS 订阅">
        <svg xmlns="http://www.w3.org/2000/svg" height="20" viewBox="0 -960 960 960" width="20"><path d="M198-120q-25.846 0-44.23-18.384-18.384-18.385-18.384-44.23 0-25.846 18.384-44.23 18.384-18.385 44.23-18.385 25.846 0 44.23 18.385 18.384 18.384 18.384 44.23 0 25.845-18.384 44.23Q223.846-120 198-120Zm538.385 0q-18.846 0-32.923-13.769-14.076-13.769-15.922-33.23-8.692-100.616-51.077-188.654-42.385-88.039-109.885-155.539-67.5-67.501-155.539-109.885Q283-663.462 182.385-672.154q-19.461-1.846-33.23-16.23-13.769-14.385-13.769-33.846t14.076-32.922q14.077-13.461 32.923-12.23 120.076 8.692 226.038 58.768 105.961 50.077 185.73 129.846 79.769 79.769 129.846 185.731 50.077 105.961 58.769 226.038 1.231 18.846-12.538 32.922Q756.461-120 736.385-120Zm-252 0q-18.231 0-32.423-13.461t-18.653-33.538Q418.155-264.23 348.886-333.5q-69.27-69.27-166.501-84.423-20.077-4.462-33.538-18.961-13.461-14.5-13.461-33.346 0-19.076 13.884-33.23 13.884-14.153 33.115-10.922 136.769 15.384 234.384 112.999 97.615 97.615 112.999 234.384 3.231 19.23-10.538 33.115Q505.461-120 484.385-120Z"/></svg>
      </a>
    
    <div id="nav-menu-btn" class="nav-icon">
      <svg xmlns="http://www.w3.org/2000/svg" height="20" viewBox="0 -960 960 960" width="20"><path d="M177.37-252.282q-17.453 0-29.477-11.964-12.024-11.963-12.024-29.326t12.024-29.537q12.024-12.174 29.477-12.174h605.26q17.453 0 29.477 11.964 12.024 11.963 12.024 29.326t-12.024 29.537q-12.024 12.174-29.477 12.174H177.37Zm0-186.218q-17.453 0-29.477-11.963-12.024-11.964-12.024-29.326 0-17.363 12.024-29.537T177.37-521.5h605.26q17.453 0 29.477 11.963 12.024 11.964 12.024 29.326 0 17.363-12.024 29.537T782.63-438.5H177.37Zm0-186.217q-17.453 0-29.477-11.964-12.024-11.963-12.024-29.326t12.024-29.537q12.024-12.174 29.477-12.174h605.26q17.453 0 29.477 11.964 12.024 11.963 12.024 29.326t-12.024 29.537q-12.024 12.174-29.477 12.174H177.37Z"/></svg>
    </div>
  </nav>
</navbar>
<div id="nav-dropdown" class="hidden">
  <div id="dropdown-link-list">
    
      <a class="nav-dropdown-link" href="/">Home</a>
    
      <a class="nav-dropdown-link" href="/archives">Archives</a>
    
      <a class="nav-dropdown-link" href="/about">About</a>
    
    
      <a class="nav-dropdown-link" href="/atom.xml" title="RSS 订阅">RSS</a>
     
    </div>
</div>
<script>
  let dropdownBtn = document.getElementById("nav-menu-btn");
  let dropdownEle = document.getElementById("nav-dropdown");
  dropdownBtn.onclick = function() {
    dropdownEle.classList.toggle("hidden");
  }
</script>
    </div>
    <div id="sidebar-wrapper">
      <sidebar id="sidebar">
  
    <div class="widget-wrap">
  <div class="info-card">
    <div class="avatar">
      
        <image src=/avatar.jpg></image>
      
      <div class="img-dim"></div>
    </div>
    <div class="info">
      <div class="username">Jichen Wen </div>
      <div class="dot"></div>
      <div class="subtitle">A PhD Student at BIG.CAS </div>
      <div class="link-list">
        
          <a class="link-btn" target="_blank" rel="noopener" href="https://twitter.com/JichenWen0725" title="Twitter"><i class="fa-brands fa-twitter"></i></a>
        
          <a class="link-btn" target="_blank" rel="noopener" href="https://steamcommunity.com/profiles/76561199384309545/" title="Steam"><i class="fa-brands fa-steam"></i></a>
        
          <a class="link-btn" target="_blank" rel="noopener" href="https://github.com/Antidington" title="GitHub"><i class="fa-brands fa-github"></i></a>
        
          <a class="link-btn" target="_blank" rel="noopener" href="https://space.bilibili.com/13424576" title="bilibili"><i class="fa-brands fa-bilibili"></i></a>
         
      </div>  
    </div>
  </div>
</div>

  
  <div class="sticky">
    
      


  <div class="widget-wrap">
    <div class="widget">
      <h3 class="widget-title">分类</h3>
      <div class="category-box">
            <a class="category-link" href="/categories/Journal-Club/">
                Journal Club
                <div class="category-count">2</div>
            </a>
        
            <a class="category-link" href="/categories/Yanqi-Notebook/">
                Yanqi Notebook
                <div class="category-count">2</div>
            </a>
        <div class="children"><div class="category-box">
            <a class="category-link" href="/categories/Yanqi-Notebook/Epigentics/">
                Epigentics
                <div class="category-count">1</div>
            </a>
        </div></div>
            <a class="category-link" href="/categories/%E5%AE%9E%E7%94%A8%E5%B7%A5%E5%85%B7/">
                实用工具
                <div class="category-count">2</div>
            </a>
        
            <a class="category-link" href="/categories/Hexo/">
                Hexo
                <div class="category-count">2</div>
            </a>
        
            <a class="category-link" href="/categories/Globus%E5%A5%97%E4%BB%B6/">
                Globus套件
                <div class="category-count">1</div>
            </a>
        
            <a class="category-link" href="/categories/Statistics/">
                Statistics
                <div class="category-count">1</div>
            </a>
        
            <a class="category-link" href="/categories/Biotools/">
                Biotools
                <div class="category-count">1</div>
            </a>
        
            <a class="category-link" href="/categories/Code-Replication/">
                Code Replication
                <div class="category-count">1</div>
            </a>
        
            <a class="category-link" href="/categories/%E5%AE%B9%E5%99%A8%E6%8A%80%E6%9C%AF/">
                容器技术
                <div class="category-count">1</div>
            </a>
        </div>
    </div>
  </div>


    
      
  <div class="widget-wrap">
    <div class="widget">
      <h3 class="widget-title">标签</h3>
      <ul class="widget-tag-list" itemprop="keywords"><li class="widget-tag-list-item"><a class="widget-tag-list-link" href="/tags/%E5%AD%A6%E4%B9%A0%E7%AC%94%E8%AE%B0/" rel="tag">学习笔记</a></li><li class="widget-tag-list-item"><a class="widget-tag-list-link" href="/tags/%E8%AF%BE%E9%A2%98%E7%9B%B8%E5%85%B3/" rel="tag">课题相关</a></li><li class="widget-tag-list-item"><a class="widget-tag-list-link" href="/tags/%E8%BD%AF%E4%BB%B6%E4%BD%BF%E7%94%A8/" rel="tag">软件使用</a></li></ul>
    </div>
  </div>


    
      
  <div class="widget-wrap">
    <div class="widget">
      <h3 class="widget-title">归档</h3>
      
      
        <a class="archive-link" href="/archives/2023/11 ">
          十一月 2023 
          <div class="archive-count">4 </div>
        </a>
      
        <a class="archive-link" href="/archives/2023/10 ">
          十月 2023 
          <div class="archive-count">9 </div>
        </a>
      
    </div>
  </div>


    
      
  <div class="widget-wrap">
    <div class="widget">
      <h3 class="widget-title">最新文章</h3>
      <ul>
        
          <a class="recent-link" href="/2023/11/05/Autobangumi_usage/" title="Autobangumi Usage for Mac" >
            <div class="recent-link-text">
              Autobangumi Usage for Mac
            </div>
          </a>
        
          <a class="recent-link" href="/2023/11/04/auto_ssh/" title="堡垒机自动登录脚本" >
            <div class="recent-link-text">
              堡垒机自动登录脚本
            </div>
          </a>
        
          <a class="recent-link" href="/2023/11/04/10yearsGWASbyBroadInstitute/" title="After a decade of genome-wide association studies, a new phase of discovery pushes on" >
            <div class="recent-link-text">
              After a decade of genome-wide association studies, a new phase of discovery pushes on
            </div>
          </a>
        
          <a class="recent-link" href="/2023/11/04/singularity_usage/" title="Singularity Usage" >
            <div class="recent-link-text">
              Singularity Usage
            </div>
          </a>
        
          <a class="recent-link" href="/2023/10/16/seqkit_usage/" title="Seqkit Usage" >
            <div class="recent-link-text">
              Seqkit Usage
            </div>
          </a>
        
      </ul>
    </div>
  </div>

    
  </div>
</sidebar>
    </div>
    <div id="content-body">
       


<article id="post-seqkit_usage" class="h-entry article article-type-post" itemprop="blogPost" itemscope itemtype="https://schema.org/BlogPosting">
  
    
   
  <div class="article-inner">
    <div class="article-main">
      <header class="article-header">
        
<div class="main-title-bar">
  <div class="main-title-dot"></div>
  
    
      <h1 class="p-name article-title" itemprop="headline name">
        Seqkit Usage
      </h1>
    
  
</div>

        <div class='meta-info-bar'>
          <div class="meta-info">
  <time class="dt-published" datetime="2023-10-15T19:07:08.000Z" itemprop="datePublished">2023-10-16</time>
</div>
          <div class="need-seperator meta-info">
            <div class="meta-cate-flex">
  
  <a class="meta-cate-link" href="/categories/Biotools/">Biotools</a>
   
</div>
  
          </div>
          <div class="wordcount need-seperator meta-info">
            9.4k 词 
          </div>
        </div>
        
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/%E8%BD%AF%E4%BB%B6%E4%BD%BF%E7%94%A8/" rel="tag">软件使用</a></li></ul>

      </header>
      <div class="e-content article-entry" itemprop="articleBody">
        
          <h1 id="seqkit-usage"><a class="markdownIt-Anchor" href="#seqkit-usage"></a> seqkit usage</h1>
<blockquote>
<p>Repo直达 <a target="_blank" rel="noopener" href="https://github.com/shenwei356/seqkit">https://github.com/shenwei356/seqkit</a></p>
</blockquote>
<h2 id="introduction"><a class="markdownIt-Anchor" href="#introduction"></a> Introduction</h2>
<p><code>seqkit</code>是重庆医科大学病毒性肝炎研究所副研究员沈伟老师基于Go语言开发的跨平台的、快速的、全面的fasta/q处理工具</p>
<h2 id="installation"><a class="markdownIt-Anchor" href="#installation"></a> Installation</h2>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># General</span></span><br><span class="line">conda install -c biconda seqkit</span><br><span class="line"><span class="comment"># Alternative for mac</span></span><br><span class="line">brew install seqkit</span><br></pre></td></tr></table></figure>
<h2 id="usage"><a class="markdownIt-Anchor" href="#usage"></a> Usage</h2>
<blockquote>
<p>Example file: AARS.lncRPI.bed<br />
Description: AARS蛋白结合lncRNA位点信息<br />
Source: <a target="_blank" rel="noopener" href="http://111.198.139.65/">POSTAR3</a> from LuLab@Tsinghua Univeristy</p>
</blockquote>
<h3 id="subseq"><a class="markdownIt-Anchor" href="#subseq"></a> subseq</h3>
<p>Get subsequences by region/gtf/bed, including flanking sequences.</p>
<blockquote>
<p>Usage:<br />
seqkit subseq [flags]</p>
<p>Flags:<br />
–bed string        by tab-delimited BED file<br />
–chr strings       select limited sequence with sequence IDs when using --gtf or --bed (multiple<br />
value supported, case ignored)<br />
-d, --down-stream int   down stream length<br />
–feature strings   select limited feature types (multiple value supported, case ignored, only<br />
works with GTF)<br />
–gtf string        by GTF (version 2.2) file<br />
–gtf-tag string    output this tag as sequence comment (default “gene_id”)<br />
-h, --help              help for subseq<br />
-f, --only-flank        only return up/down stream sequence<br />
-r, --region string     by region. e.g 1:12 for first 12 bases, -12:-1 for last 12 bases, 13:-1 for<br />
cutting first 12 bases. type “seqkit subseq -h” for more examples<br />
-u, --up-stream int     up stream length<br />
-U, --update-faidx      update the fasta index file if it exists. Use this if you are not sure whether<br />
the fasta file changed</p>
</blockquote>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">seqkit subseq --bed AARS.lncRPI.bed GRCh38.p13.genome.fa.gz &gt; AARS.lncRPI.fasta</span><br></pre></td></tr></table></figure>
<h3 id="stats"><a class="markdownIt-Anchor" href="#stats"></a> stats</h3>
<p>Simple statistics of FASTA/Q files.</p>
<blockquote>
<p>Flags:<br />
-a, --all                  all statistics, including quartiles of seq length, sum_gap, N50<br />
-b, --basename             only output basename of files<br />
-E, --fq-encoding string   fastq quality encoding. available values: ‘sanger’, ‘solexa’, ‘illumina-1.3+’, ‘illumina-1.5+’, ‘illumina-1.8+’. (default “sanger”)<br />
-G, --gap-letters string   gap letters (default “- .”)<br />
-h, --help                 help for stats<br />
-e, --skip-err             skip error, only show warning message<br />
-i, --stdin-label string   label for replacing default “-” for stdin (default “-”)<br />
-T, --tabular              output in machine-friendly tabular format</p>
</blockquote>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">seqkit <span class="built_in">stat</span> -a -T AARS.lncRPI.fasta | csvlook</span><br></pre></td></tr></table></figure>
<p><img src="image-1.png" alt="" /></p>
<h3 id="watch"><a class="markdownIt-Anchor" href="#watch"></a> watch</h3>
<p>Monitoring and online histograms of sequence features</p>
<blockquote>
<p>Usage:<br />
seqkit watch [flags]</p>
<p>Flags:<br />
-B, --bins int                  number of histogram bins (default -1)<br />
-W, --delay int                 sleep this many seconds after online plotting (default 1)<br />
-y, --dump                      print histogram data to stderr instead of plotting<br />
-f, --fields string             target fields, available values: ReadLen, MeanQual, GC, GCSkew<br />
(default “ReadLen”)<br />
-h, --help                      help for watch<br />
-O, --img string                save histogram to this PDF/image file<br />
-H, --list-fields               print out a list of available fields<br />
-L, --log                       log10(x+1) transform numeric values<br />
-x, --pass                      pass through mode (write input to stdout)<br />
-p, --print-freq int            print/report after this many records (-1 for print after EOF) (default -1)<br />
-b, --qual-ascii-base int       ASCII BASE, 33 for Phred+33 (default 33)<br />
-Q, --quiet-mode                supress all plotting to stderr<br />
-R, --reset                     reset histogram after every report<br />
-v, --validate-seq              validate bases according to the alphabet<br />
-V, --validate-seq-length int   length of sequence to validate (0 for whole seq) (default 10000)</p>
</blockquote>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">seqkit watch -f ReadLen AARS.lncRPI.fasta</span><br></pre></td></tr></table></figure>
<p><img src="image-2.png" alt="" /></p>
<h3 id="seq"><a class="markdownIt-Anchor" href="#seq"></a> seq</h3>
<p>Transform sequences (extract ID, filter by length, remove gaps, reverse complement…)</p>
<blockquote>
<p>Usage:<br />
seqkit seq [flags]</p>
<p>Flags:<br />
-k, --color                     colorize sequences - to be piped into “less -R”<br />
-p, --complement                complement sequence, flag ‘-v’ is recommended to switch on<br />
–dna2rna                   DNA to RNA<br />
-G, --gap-letters string        gap letters to be removed with -g/–remove-gaps (default “- \t.”)<br />
-h, --help                      help for seq<br />
-l, --lower-case                print sequences in lower case<br />
-M, --max-len int               only print sequences shorter than or equal to the maximum length (-1<br />
for no limit) (default -1)<br />
-R, --max-qual float            only print sequences with average quality less than this limit (-1 for<br />
no limit) (default -1)<br />
-m, --min-len int               only print sequences longer than or equal to the minimum length (-1<br />
for no limit) (default -1)<br />
-Q, --min-qual float            only print sequences with average quality qreater or equal than this<br />
limit (-1 for no limit) (default -1)<br />
-n, --name                      only print names/sequence headers<br />
-i, --only-id                   print IDs instead of full headers<br />
-q, --qual                      only print qualities<br />
-b, --qual-ascii-base int       ASCII BASE, 33 for Phred+33 (default 33)<br />
-g, --remove-gaps               remove gaps letters set by -G/–gap-letters, e.g., spaces, tabs, and<br />
dashes (gaps “-” in aligned sequences)<br />
-r, --reverse                   reverse sequence<br />
–rna2dna                   RNA to DNA<br />
-s, --seq                       only print sequences<br />
-u, --upper-case                print sequences in upper case<br />
-v, --validate-seq              validate bases according to the alphabet<br />
-V, --validate-seq-length int   length of sequence to validate (0 for whole seq) (default 10000)</p>
</blockquote>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 查看.fasta文件</span></span><br><span class="line">seqkit <span class="built_in">seq</span> -k AARS.lncRPI.fasta</span><br></pre></td></tr></table></figure>
<p><img src="image-3.png" alt="" /></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 提取ID</span></span><br><span class="line">seqkit <span class="built_in">seq</span> AARS.lncRPI.fasta -n</span><br></pre></td></tr></table></figure>
<p><img src="image-4.png" alt="" /></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 提取序列</span></span><br><span class="line">seqkit <span class="built_in">seq</span> AARS.lncRPI.fasta -s</span><br></pre></td></tr></table></figure>
<p><img src="image-5.png" alt="" /></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 过滤长度小于40bp的序列</span></span><br><span class="line">seqkit <span class="built_in">seq</span> -m 40 -g AARS.lncRPI.fasta | seqkit stats -a -T | csvlook</span><br></pre></td></tr></table></figure>
<p><img src="image-6.png" alt="" /></p>
<h3 id="faidx"><a class="markdownIt-Anchor" href="#faidx"></a> faidx</h3>
<p>Create FASTA index file and extract subsequence</p>
<blockquote>
<p>Usage:<br />
seqkit faidx [flags] <fasta-file> [regions…]</p>
<p>Flags:<br />
-f, --full-head            print full header line instead of just ID. &gt;New fasta index file ending with<br />
.seqkit.fai will be created<br />
-h, --help                 help for faidx<br />
-i, --ignore-case          ignore case<br />
-I, --immediate-output     print output immediately, do not use write buffer<br />
-l, --region-file string   file containing a list of regions<br />
-U, --update-faidx         update the fasta index file if it exists. Use this if you are not sure<br />
whether the fasta file changed<br />
-r, --use-regexp           IDs are regular expression. But subseq region is not supported here.</p>
</blockquote>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 筛选位于chr15的序列</span></span><br><span class="line">seqkit faidx AARS.lncRPI.fasta chr15 -r -f | seqkit stats -a -T | csvlook</span><br></pre></td></tr></table></figure>
<p><img src="image-7.png" alt="" /></p>
<h3 id="locate"><a class="markdownIt-Anchor" href="#locate"></a> locate</h3>
<p>Locate subsequences/motifs, mismatch allowed</p>
<blockquote>
<p>Usage:<br />
seqkit locate [flags]</p>
<p>Flags:<br />
–bed                       output in BED6 format<br />
-c, --circular                  circular genome. type “seqkit locate -h” for details<br />
-d, --degenerate                pattern/motif contains degenerate base<br />
–gtf                       output in GTF format<br />
-h, --help                      help for locate<br />
-M, --hide-matched              do not show matched sequences<br />
-i, --ignore-case               ignore case<br />
-I, --immediate-output          print output immediately, do not use write buffer<br />
-m, --max-mismatch int          max mismatch when matching by seq. For large genomes like human<br />
genome, using mapping/alignment tools would be faster<br />
-G, --non-greedy                non-greedy mode, faster but may miss motifs overlapping with others<br />
-P, --only-positive-strand      only search on positive strand<br />
-p, --pattern strings           pattern/motif (multiple values supported. Attention: use double<br />
quotation marks for patterns containing comma, e.g., -p ‘“A{2,}”’)<br />
-f, --pattern-file string       pattern/motif file (FASTA format)<br />
-F, --use-fmi                   use FM-index for much faster search of lots of sequence patterns<br />
-r, --use-regexp                patterns/motifs are regular expression<br />
-V, --validate-seq-length int   length of sequence to validate (0 for whole seq) (default 10000)</p>
</blockquote>
<p>使用MEME得到的motif<br />
<img src="image-8.png" alt="" /></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># Locate motif found by MEME</span></span><br><span class="line">seqkit <span class="built_in">seq</span> AARS.lncRPI.fasta \</span><br><span class="line">    | seqkit locate -i -d -p CACCATATTAAAGGAGGGAGTGGCAGAAAACGTGAGGTGGG \</span><br><span class="line">    | <span class="built_in">head</span> \</span><br><span class="line">    | csvtk pretty -t</span><br><span class="line"><span class="comment"># Save the located ID</span></span><br><span class="line">seqkit <span class="built_in">seq</span> AARS.lncRPI.fasta \</span><br><span class="line">    | seqkit locate -i -d -p CACCATATTAAAGGAGGGAGTGGCAGAAAACGTGAGGTGGG \</span><br><span class="line">    | <span class="built_in">cut</span> -f 1 \</span><br><span class="line">    | gerp -v <span class="string">&quot;seqID&quot;</span> &gt; extract.id</span><br></pre></td></tr></table></figure>
<p><img src="image-9.png" alt="Alt text" /></p>
<h3 id="grep"><a class="markdownIt-Anchor" href="#grep"></a> grep</h3>
<p>Search sequences by ID/name/sequence/sequence motifs, mismatch allowed.</p>
<blockquote>
<p>Usage:<br />
seqkit grep [flags]</p>
<p>Flags:<br />
-n, --by-name                match by full name instead of just ID<br />
-s, --by-seq                 search subseq on seq, both positive and negative strand are searched, and<br />
mismatch allowed using flag -m/–max-mismatch<br />
-c, --circular               circular genome<br />
-C, --count                  just print a count of matching records. with the -v/–invert-match flag,<br />
count non-matching records<br />
-d, --degenerate             pattern/motif contains degenerate base<br />
–delete-matched         delete a pattern right after being matched, this keeps the firstly<br />
matched data and speedups when using regular expressions<br />
-h, --help                   help for grep<br />
-i, --ignore-case            ignore case<br />
-I, --immediate-output       print output immediately, do not use write buffer<br />
-v, --invert-match           invert the sense of matching, to select non-matching records<br />
-m, --max-mismatch int       max mismatch when matching by seq. For large genomes like human genome,<br />
using mapping/alignment tools would be faster<br />
-P, --only-positive-strand   only search on positive strand<br />
-p, --pattern strings        search pattern (multiple values supported. Attention: use double<br />
quotation marks for patterns containing comma, e.g., -p ‘“A{2,}”’)<br />
-f, --pattern-file string    pattern file (one record per line)<br />
-R, --region string          specify sequence region for searching. e.g 1:12 for first 12 bases,<br />
-12:-1 for last 12 bases<br />
-r, --use-regexp             patterns are regular expression</p>
</blockquote>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br></pre></td><td class="code"><pre><span class="line">seqkit grep -f extract.id AARS.lncRPI.fasta -o extract.fasta</span><br><span class="line">```  </span><br><span class="line"></span><br><span class="line"><span class="comment">### split</span></span><br><span class="line">Split sequences into files by name ID, subsequence of given region,</span><br><span class="line">part size or number of parts.</span><br><span class="line">&gt;Usage:</span><br><span class="line">&gt;  seqkit <span class="built_in">split</span> [flags]</span><br><span class="line">&gt;</span><br><span class="line">&gt;Flags:</span><br><span class="line">&gt;  -i, --by-id                     <span class="built_in">split</span> squences according to sequence ID</span><br><span class="line">      --by-id-prefix string       file prefix <span class="keyword">for</span> --by-id</span><br><span class="line">  -p, --by-part int               <span class="built_in">split</span> sequences into N parts</span><br><span class="line">      --by-part-prefix string     file prefix <span class="keyword">for</span> --by-part</span><br><span class="line">  -r, --by-region string          <span class="built_in">split</span> squences according to subsequence of given region. e.g 1:12 <span class="keyword">for</span></span><br><span class="line">                                  first 12 bases, -12:-1 <span class="keyword">for</span> last 12 bases. <span class="built_in">type</span> <span class="string">&quot;seqkit split -h&quot;</span> <span class="keyword">for</span></span><br><span class="line">                                  more examples</span><br><span class="line">      --by-region-prefix string   file prefix <span class="keyword">for</span> --by-region</span><br><span class="line">  -s, --by-size int               <span class="built_in">split</span> sequences into multi parts with N sequences</span><br><span class="line">      --by-size-prefix string     file prefix <span class="keyword">for</span> --by-size</span><br><span class="line">  -d, --dry-run                   dry run, just <span class="built_in">print</span> message and no files will be created.</span><br><span class="line">  -e, --extension string          <span class="built_in">set</span> output file extension, e.g., <span class="string">&quot;.gz&quot;</span>, <span class="string">&quot;.xz&quot;</span>, or <span class="string">&quot;.zst&quot;</span></span><br><span class="line">  -f, --force                     overwrite output directory</span><br><span class="line">  -h, --<span class="built_in">help</span>                      <span class="built_in">help</span> <span class="keyword">for</span> <span class="built_in">split</span></span><br><span class="line">  -k, --keep-temp                 keep temporary FASTA and .fai file when using 2-pass mode</span><br><span class="line">  -O, --out-dir string            output directory (default value is <span class="variable">$infile</span>.<span class="built_in">split</span>)</span><br><span class="line">  -2, --two-pass                  two-pass mode <span class="built_in">read</span> files twice to lower memory usage. (only <span class="keyword">for</span> FASTA</span><br><span class="line">                                  format)</span><br><span class="line">  -U, --update-faidx              update the fasta index file <span class="keyword">if</span> it exists. Use this <span class="keyword">if</span> you are not sure</span><br><span class="line">                                  whether the fasta file changed</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">```bash</span><br><span class="line"><span class="comment"># 分割fasta，每个fasta 1000条序列</span></span><br><span class="line">seqkit <span class="built_in">split</span> AARS.lncRPI.fasta -s 1000</span><br><span class="line">[INFO] <span class="built_in">split</span> into 1000 seqs per file</span><br><span class="line">[INFO] write 1000 sequences to file: AARS.lncRPI.fasta.split/AARS.lncRPI.part_001.fasta</span><br><span class="line">[INFO] write 317 sequences to file: AARS.lncRPI.fasta.split/AARS.lncRPI.part_002.fasta</span><br><span class="line">seqkit stats -a -T AARS.lncRPI.fasta.split/* | csvlook</span><br></pre></td></tr></table></figure>
<p><img src="image-10.png" alt="" /></p>
<h3 id="shuffle"><a class="markdownIt-Anchor" href="#shuffle"></a> shuffle</h3>
<p>Shuffle sequences</p>
<blockquote>
<p>Usage:<br />
seqkit shuffle [flags]</p>
<p>Flags:<br />
-h, --help            help for shuffle<br />
-k, --keep-temp       keep temporary FASTA and .fai file when using 2-pass mode<br />
-s, --rand-seed int   rand seed for shuffle (default 23)<br />
-2, --two-pass        two-pass mode read files twice to lower memory usage. (only for FASTA format)<br />
-U, --update-faidx    update the fasta index file if it exists. Use this if you are not sure whether<br />
the fasta file changed</p>
</blockquote>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">seqkit shuffle AARS.lncRPI.fasta &gt; shuffle.fasta</span><br><span class="line">[INFO] <span class="built_in">read</span> sequences ...</span><br><span class="line">[INFO] 1317 sequences loaded</span><br><span class="line">[INFO] shuffle ...</span><br><span class="line">[INFO] output ...</span><br></pre></td></tr></table></figure>
<p><img src="image-11.png" alt="" /></p>
<h2 id="reference"><a class="markdownIt-Anchor" href="#reference"></a> Reference</h2>
<ol>
<li><a target="_blank" rel="noopener" href="https://bioinf.shenwei.me/seqkit/">SeqKit - a cross-platform and ultrafast toolkit for FASTA/Q file manipulation</a></li>
</ol>

        
      </div>

         
    </div>
    
     
  </div>
  
    
<nav id="article-nav">
  <a class="article-nav-btn left "
    
      href="/2023/11/04/singularity_usage/"
      title="Singularity Usage"
     >
    <i class="fa-solid fa-angle-left"></i>
    <p class="title-text">
      
        Singularity Usage
        
    </p>
  </a>
  <a class="article-nav-btn right "
    
      href="/2023/10/14/GlobusConnectUsage/"
      title="Globus Connect Usage"
     >

    <p class="title-text">
      
        Globus Connect Usage
        
    </p>
    <i class="fa-solid fa-angle-right"></i>
  </a>
</nav>


  
</article>




    </div>
    <div id="footer-wrapper">
      <footer id="footer">
  
  <div id="footer-info" class="inner">
    
    &copy; 2023 Jichen Wen<br>
    Powered by <a href="https://hexo.io/" target="_blank">Hexo</a> & Theme <a target="_blank" rel="noopener" href="https://github.com/saicaca/hexo-theme-vivia">Vivia</a>
  </div>
</footer>

    </div>
    <div class="back-to-top-wrapper">
    <button id="back-to-top-btn" class="back-to-top-btn hide" onclick="topFunction()">
        <i class="fa-solid fa-angle-up"></i>
    </button>
</div>

<script>
    function topFunction() {
        window.scroll({ top: 0, behavior: 'smooth' });
    }
    let btn = document.getElementById('back-to-top-btn');
    function scrollFunction() {
        if (document.body.scrollTop > 600 || document.documentElement.scrollTop > 600) {
            btn.classList.remove('hide')
        } else {
            btn.classList.add('hide')
        }
    }
    window.onscroll = function() {
        scrollFunction();
    }
</script>

  </div>
  <script src="/js/light-dark-switch.js"></script>
</body>
</html>
